modules/compute/src/main/java/org/apache/ignite/internal/compute/ExecutionManager.java

```diff
@@ -123,7 +123,7 @@ public CompletableFuture<List<JobState>> localStatesAsync() {
      */
     public CompletableFuture<@Nullable JobState> stateAsync(UUID jobId) {
         JobExecution<?> execution = executions.get(jobId);
-        if (execution != null) {
+        if (execution != null && !(execution instanceof RemoteJobExecution)) {
```

Comment:
Maybe we should add the clarification here why we have this creepy `instanceof`.